lib/pbench/test/unit/server/auth/test_auth.py

```diff
@@ -415,6 +415,24 @@ def test_token_introspect_sig(self, monkeypatch, rsa_keys):
             str(exc.value.__cause__) == "Signature verification failed"
         ), f"{exc.value.__cause__}"
 
+    def test_token_introspect_alg(self, monkeypatch, rsa_keys):
+        """Verify .token_introspect_offline() failure via algorithm error"""
+        client_id = "us"
+        generated_api_key = jwt.encode(
+            {"some_key": "some_value"}, "my_secret", algorithm="HS256"
+        )
+        config = mock_connection(
+            monkeypatch, client_id, public_key=rsa_keys["public_key"]
+        )
+        oidc_client = OpenIDClient.construct_oidc_client(config)
+
+        with pytest.raises(OpenIDTokenInvalid) as exc:
+            # Make the signature invalid.
+            oidc_client.token_introspect(generated_api_key)
```

Comment:
That comment reads funny.  Shouldn't it be applied to line 421 instead of here?

Reply:
😆 You corrected the contents of the comment (which I'd failed to notice), but I still think it's marking the wrong line:  there is nothing wrong with line 431; what's "wrong" is the way the API key was generated, which happened several lines above this point.  (If you want to have a comment explaining why we use `pytest.raises()` it should probably be placed before we make that call, and you might want to elaborate a bit...but, I think it would be fine if you just moved this existing comment up to where you specify the algorithm.)

Reply:
Yeah you are right, it should be on line 421